doc/modules/neural_networks.rst

```diff
@@ -1,5 +1,131 @@
 .. _neural_network:
 
+==================================
+Neural network models (supervised)
+==================================
+
+.. currentmodule:: sklearn.neural_network
+
+.. _random_basis_function:
+
+Random basis function
+=====================
+
+Random basis function :math: `f(X): R \rightarrow R` that projects matrix
+:math: `X` into another feature space where the number of features is less, equal
+or higher than the original feature space. The output matrix :math: `H` is
+computed as follows:
+
+.. math::
+
+   H = g(Xw + b)
+
+where :math: `g(\cdot): R \rightarrow R` is the activation function, :math: `w`
+is the weight parameter vector, and :math: `b` is the intercept vector.
+
+:math: `w \in R^{d \times k}`, and :math: `b \in R^{d}` are generated based
+on the uniform distribution scaled between two values, set by the user.
+
+
+The example code below illustrates using this function::
+
+    >>> from sklearn.neural_network import RandomBasisFunction
+    >>> X = [[0, 0], [1, 1]]
+    >>> rbf = RandomBasisFunction(random_state=1, n_activated_features=2)
+    >>> rbf.fit(X, y)
```

Comment:
That's true, not sure what else to call it :(. I will call it `fe` as short for "feature extractor".